airflow-core/pyproject.toml

```diff
@@ -94,6 +94,7 @@ dependencies = [
     'libcst >=1.8.2',
     "linkify-it-py>=2.0.0",
     "lockfile>=0.12.2",
+    "memray>=1.19.0",
```

Comment:
I think it would be better to have it an optional "memray" dependency. We do not necessarily want to have it always installed.

Reply:
Why ? It's like 2.4 MB size, and also potentially difficult to install for various system/python version - it's native library and we do not want to make it a requirement.